pkg/cmd/pr/create/create.go

```diff
@@ -659,64 +588,110 @@ func NewCreateContext(opts *CreateOptions) (*CreateContext, error) {
+	// --head or the current branch.
+	var targetHeadBranch string
+	var targetHeadRepoOwner string
+
+	promptForHeadRepo := true
+
+	if opts.HeadBranch != "" {
+		promptForHeadRepo = false
+		targetHeadBranch = opts.HeadBranch
+		// If the --head provided contains a colon, that means
+		// this is <owner>:<branch> syntax.
+		if idx := strings.IndexRune(opts.HeadBranch, ':'); idx >= 0 {
+			targetHeadRepoOwner = opts.HeadBranch[:idx]
+			targetHeadBranch = opts.HeadBranch[idx+1:]
+		}
+	} else {
+		// Use the current branch as the target local head branch when
+		// --head is not provided.
+		targetHeadBranch, err = opts.Branch()
 		if err != nil {
 			return nil, fmt.Errorf("could not determine the current branch: %w", err)
 		}
-		headBranchLabel = headBranch
-		isPushEnabled = true
-	} else if idx := strings.IndexRune(headBranch, ':'); idx >= 0 {
-		headBranch = headBranch[idx+1:]
 	}
 
-	gitClient := opts.GitClient
-	if ucc, err := gitClient.UncommittedChangeCount(context.Background()); err == nil && ucc > 0 {
-		fmt.Fprintf(opts.IO.ErrOut, "Warning: %s\n", text.Pluralize(ucc, "uncommitted change"))
+	targetHeadBranchConfig, err := gitClient.ReadBranchConfig(context.Background(), targetHeadBranch)
+	if err != nil {
+		return nil, err
 	}
 
-	var headRepo ghrepo.Interface
-	var headRemote *ghContext.Remote
+	// See if we can determine if this branch has been push previously with
+	// Git configurations and @{push} revision syntax.
+	remotePushDefault, err := opts.GitClient.RemotePushDefault(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// Suppressing these errors as we have other means of computing the PullRequestRefs when these fail.
+	parsedPushRevision, _ := opts.GitClient.ParsePushRevision(ctx, targetHeadBranch)
+	pushDefault, err := opts.GitClient.PushDefault(ctx)
+	if err != nil {
+		return nil, err
+	}
 
-	headBranchConfig, err := gitClient.ReadBranchConfig(context.Background(), headBranch)
+	prRefs, err := shared.ParsePRRefs(targetHeadBranch, targetHeadBranchConfig, parsedPushRevision, pushDefault, remotePushDefault, targetBaseRepo, remotes)
 	if err != nil {
 		return nil, err
 	}
-	if isPushEnabled {
-		// TODO: This doesn't respect the @{push} revision resolution or triagular workflows assembled with
-		// remote.pushDefault, or branch.<branchName>.pushremote config settings. The finder's ParsePRRefs
-		// may be able to replace this function entirely.
-		if trackingRef, found := tryDetermineTrackingRef(gitClient, remotes, headBranch, headBranchConfig); found {
-			isPushEnabled = false
-			if r, err := remotes.FindByName(trackingRef.remoteName); err == nil {
-				headRepo = r
-				headRemote = r
-				headBranchLabel = trackingRef.branchName
-				if !ghrepo.IsSame(baseRepo, headRepo) {
-					headBranchLabel = fmt.Sprintf("%s:%s", headRepo.RepoOwner(), trackingRef.branchName)
+
+	// If the --head provided contains <repo_name>:<branch>  syntax, we need to use
+	// the provided owner instead of the owner of the base repository.
```

Comment:
`gh pr create` only supports user-based forks, not organization-based forks.  So my only concern with using the `owner` language is that it could be misleading as organizations and users can both be owners. $0.02

```suggestion
	// If the --head provided contains <user>:<branch> syntax, we need to use
	// the provided owner instead of the owner of the base repository.
```

Reply:
fixed in 7f47e7087a17250e9bb6eeffe64afe001821125e